File: scientific-calculator/calculator.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

func Calculate(op string, a, b float64) float64 {
	switch op {
	case "add":
		return a + b
	case "subtract":
		return a - b
	case "multiply":
		return a * b
	case "divide":
		return a / b
	case "power":
		return math.Pow(a, b)
	case "sqrt":
		return math.Sqrt(a)
	case "sin":
		return math.Sin(a)
	case "cos":
		return math.Cos(a)
	case "tan":
		return math.Tan(a)
	case "log":
		return math.Log10(a)
	case "ln":
		return math.Log(a)
	default:
		panic("Unknown operation: " + op)
	}
}

func main() {
	if len(os.Args) < 3 {
		fmt.Println("Usage: go run calculator.go <op> <a> [b]")
		fmt.Println("Operations: add, subtract, multiply, divide, power, sqrt, sin, cos, tan, log, ln")
		fmt.Println("Examples:")
		fmt.Println("  go run calculator.go add 2 3")
		fmt.Println("  go run calculator.go sqrt 16")
		fmt.Println("  go run calculator.go sin 1.5708")
		os.Exit(1)
	}

	op := os.Args[1]
	a, err := strconv.ParseFloat(os.Args[2], 64)
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error: invalid number for a")
		os.Exit(1)
	}

	var b float64
	if len(os.Args) > 3 {
		b, err = strconv.ParseFloat(os.Args[3], 64)
		if err != nil {
			fmt.Fprintln(os.Stderr, "Error: invalid number for b")
			os.Exit(1)
		}
	}

	defer func() {
		if r := recover(); r != nil {
			fmt.Fprintln(os.Stderr, "Error:", r)
			os.Exit(1)
		}
	}()

	result := Calculate(op, a, b)
	fmt.Println(result)
}

File: scientific-calculator/calculator.ts
export function calculate(op: string, a: number, b?: number): number {
  switch (op) {
    case "add": return a + b!;
    case "subtract": return a - b!;
    case "multiply": return a * b!;
    case "divide": return a / b!;
    case "power": return Math.pow(a, b!);
    case "sqrt": return Math.sqrt(a);
    case "sin": return Math.sin(a);
    case "cos": return Math.cos(a);
    case "tan": return Math.tan(a);
    case "log": return Math.log10(a);
    case "ln": return Math.log(a);
    default: throw new Error(`Unknown operation: ${op}`);
  }
}

function main() {
  const args = process.argv.slice(2);
  if (args.length < 2) {
    console.log("Usage: node calculator.js <op> <a> [b]");
    console.log("Operations: add, subtract, multiply, divide, power, sqrt, sin, cos, tan, log, ln");
    console.log("Examples:");
    console.log("  node calculator.js add 2 3");
    console.log("  node calculator.js sqrt 16");
    console.log("  node calculator.js sin 1.5708");
    process.exit(1);
  }

  const op = args[0];
  const a = parseFloat(args[1]);
  const b = args.length > 2 ? parseFloat(args[2]) : undefined;

  try {
    const result = calculate(op, a, b);
    console.log(result);
  } catch (e) {
    console.error("Error:", (e as Error).message);
    process.exit(1);
  }
}

if (require.main === module) {
  main();
}